src/CcdImage.cc

```diff
@@ -41,6 +41,8 @@ static double sq(const double &x) { return x*x;}
 
 void CcdImage::LoadCatalog(const lsst::afw::table::SortedCatalogT<lsst::afw::table::SourceRecord> &Cat,const std::string &fluxField)
 {
+  // TODO: these should use slots, e.g. "slot_Centroid_x", or better:
+  // self.config.centroidName + "_x" in python.
```

Comment:
Isn't this a place that you should be using the slot?  The point of slots is to provide defaults for e.g. centroiders --- if you want to use a different algorithm you should change the slot not configure jointcal.